pkg/controller/keycloak/keycloak_reconciler.go

```diff
@@ -283,3 +289,14 @@ func (i *KeycloakReconciler) getKeycloakIngressDesiredState(clusterState *common
 		Msg: "Update Keycloak Ingress",
 	}
 }
+
+func (i *KeycloakReconciler) getPostgresqlServiceEndpointsDesiredState(clusterState *common.ClusterState, cr *kc.Keycloak) common.ClusterAction {
+	if clusterState.PostgresqlServiceEndpoints == nil {
+		// This happens only during initial run
+		return nil
+	}
+	return common.GenericUpdateAction{
```

Comment:
How come there is no create action here and only an update?

Reply:
The `Endpoints` objects is created and maintained by Kubernetes in general. Typically, they shouldn't be touched by the users, however, redirecting traffic to an external thing is one of such use cases. 

You may find more information about this approach on OpenShift Documentation (1). However, the latest version of the documentation has been restructured and this page has been wiped out. 

(1) https://docs.openshift.com/container-platform/3.11/dev_guide/integrating_external_services.html

Reply:
ah ok, perfect. Thank you.